### PR TITLE
Use VideoFrames as input for VideoDecoders

### PR DESCRIPTION
**Pull request by murillo128**
Yakes over the work done in https://github.com/medooze/media-server/pull/271

### CODE REVIEW

**Review by mildsunrise — APPROVED** [review]
LGTM, but when taking over a PR I think it's a good idea to either work on the same PR, or at least keep the history so that the additional changes (by you) are easier to see :pray:

**Comment by mildsunrise** [comment]
I see you also adapted https://github.com/medooze/video-codecs-node/pull/9 to be compatible with the new changes, thanks!

File: src/VideoDecoderWorker.cpp [hunk 1]
```diff
@@ -80,191 +80,117 @@ void VideoDecoderWorker::RemoveVideoOutput(VideoOutput* output)
...
-		DWORD clockRate = packet->GetClockRate();
-		QWORD senderTime = packet->HasAbsoluteCaptureTime() ? packet->GetAbsoluteCaptureTime() : 0;
-
 		//If we don't have codec
-		if (!videoDecoder || (packet->GetCodec()!=videoDecoder->type))
+		if (!videoDecoder || (videoFrame->GetCodec()!=videoDecoder->type))
 		{
 			//Create new codec from pacekt
-			videoDecoder.reset(VideoCodecFactory::CreateDecoder((VideoCodec::Type)packet->GetCodec()));
+			videoDecoder.reset(VideoCodecFactory::CreateDecoder((VideoCodec::Type)videoFrame->GetCodec()));
```
Comment (mildsunrise) [inline src/VideoDecoderWorker.cpp:107]: cast no longer needed, better remove it

```suggestion
			videoDecoder.reset(VideoCodecFactory::CreateDecoder(videoFrame->GetCodec()));
```

File: src/VideoDecoderWorker.cpp [hunk 2]
```diff
@@ -274,21 +200,13 @@ int VideoDecoderWorker::Decode()
...
 }
 
-void VideoDecoderWorker::onRTP(const RTPIncomingMediaStream* stream, const RTPPacket::shared& packet)
+void VideoDecoderWorker::onMediaFrame(const MediaFrame& frame)
 {
-	//Put it on the queue
-	packets.Add(packet->Clone());
-}
+	//Ensure it is video
+	if (frame.GetType()!=MediaFrame::Video)
```
Comment (mildsunrise) [inline src/VideoDecoderWorker.cpp:206]: hmm, I see you removed the Warning() here, any particular reason?

**Review by bcostdolby — APPROVED** [review]
lgtm

**Comment by murillo128** [comment]
we have a problem with ffmpeg expecting annexb h26X streams while we are feeding non-annexb with the default packetizers.

File: src/VideoDecoderWorker.cpp (hunk 2 above)
Comment (murillo128) [reply src/VideoDecoderWorker.cpp:179]: I started from scratch, so probably miss it. Happy to add it back.

**Pull request merged**